### PR TITLE
Remove tox and adjust markdown files

### LINKED ISSUE #1157: remove tox references and update README
README should not contain a call to `tox -e docs` but rather the updated sphinx commands.

Also tox.ini and references to tox should be removed as it's not used anymore

### PR DESCRIPTION
**Pull request by mgiulini**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md).

## Checklist

- [ ] Tests added for the new code
- [ ] Documentation added for the code changes
- [ ] Does not break licensing
- [ ] Does not add any dependencies, if it does please add a thorough explanation

## Related Issue
Closes #1157 by eliminating references to tox and updating the content of contributing.md file.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,52 +25,27 @@ To contribute to the HADDOCK3's Python shell, follow these steps:
...
-    [pytest](https://docs.pytest.org/en/6.2.x/) for testing.
-    3. See more details on how to contribute with code and tests in the
+    2. See more details on how to contribute with code and tests in the
     subheadings below.
 
-4.  While you are developing (or when you think you have finished), you can
-    (should) use our `tox` environments to test your code. Use the
-    following commands from the main repository folder:
+4.  Test your code:
+    1.  Tests sit in the `tests/` folder. Use
```
Comment (VGPReys) [inline CONTRIBUTING.md:36]:
```suggestion
    1.  Unity-tests sit in the `tests/` folder. Use
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -302,9 +175,17 @@ the documentation website.
 
 You can render the documentation locally to inspect the end result
 before creating a pull request. To compile the documentation locally:
-activate the `haddock3` python environment inside the haddock3 github
-folder, run `tox -e docs` and then use your favourite browser to open the
-`haddock3-docs/index.html` file. We invite you to read through Sphinx-doc
+- install HADDOCK in development mode with the `docs` extra requirements
+  (`pip install -e '.[docs]'`)
```
Comment (VGPReys) [inline CONTRIBUTING.md:179]:
```suggestion
- install HADDOCK in development mode with the `dev` and `docs` extra requirements
  (`pip install -e '.[dev,docs]'`)
```

Comment (mgiulini) [reply CONTRIBUTING.md:179]: no need to install dev requirements to render the documentation

File: README.md [hunk 3]
```diff
@@ -103,10 +103,17 @@ Please check [DEVELOPMENT](https://github.com/haddocking/haddock3/blob/main/DEVE
 
 The code documentation is automatically built and hosted at [bonvinlab.org/haddock3](https://www.bonvinlab.org/haddock3/).
 
-To build it locally (considering you have followed the installation instructions):
+To build it locally it is necessary to have some extra packages installed. You can install them using the following command:
 
 ```bash
-tox -e docs
+pip install -e '.[docs]'
```
[the remark this answers is not in the document]
Comment (mgiulini) [reply README.md:109]: same as above

File: CONTRIBUTING.md (hunk 1 above)
Comment (mgiulini) [reply CONTRIBUTING.md]: It's "unit tests" :)

File: CONTRIBUTING.md
Comment (VGPReys) [inline CONTRIBUTING.md:1]: L.21, `tox` is still mentioned.
L.49, `tox` is still mentioned.

There is no instruction on how to install the `dev` env.
```bash
pip install '.[dev]'
```

Comment (mgiulini) [reply CONTRIBUTING.md:1]: removed tox refs!

Comment (mgiulini) [reply CONTRIBUTING.md:1]: I would not add the dev env instructions..we clearly say that before contributing one should look at the INSTALL file

**Review by VGPReys — APPROVED** [review]

**Pull request merged**